solr/core/src/test/org/apache/solr/cloud/MultiThreadedOCPTest.java

```diff
@@ -50,15 +50,7 @@ public MultiThreadedOCPTest() {
   }
 
   @Test
-  public void test() throws Exception {
-    testParallelCollectionAPICalls();
-    testTaskExclusivity();
-    testDeduplicationOfSubmittedTasks();
-    testLongAndShortRunningParallelApiCalls();
-    testFillWorkQueue();
```

Comment:
Turning the five private methods into top-level tests means they could run in different orders, so more realistic test coverage there as a side effect, nice.

From a quick read of the tests, they don't look like one could cause side effects that would change the baseline for the next tests. So changing order should be fine.

Am less sure though on whether or not the tests could run in parallel now then and if then they could interact with each other? The ones which create collections, the collection names are specific to each test, so no problem there at least.

Reply:
`MAX_PARALLEL_TASKS` in `testFillWorkQueue()` might have a cross test impact? When the queue is full, it's full.

Reply:
I guess it might just delay the other tests if running concurrently. Let's leave it like this and see if new problems appear?

Reply:
I don't _think_ that we run tests from the same class concurrently in the same JVM, but that also seems like an implementation detail that we shouldn't rely on. Will watch for breakage.